include/swift/Frontend/InputFile.h

```diff
@@ -41,7 +41,9 @@ class InputFile {
   /// string.
   InputFile(StringRef name, bool isPrimary,
             llvm::MemoryBuffer *buffer = nullptr)
-      : Filename(name), IsPrimary(isPrimary), Buffer(buffer) {
+      : Filename(
+            convertBufferNameFromLLVM_getFileOrSTDIN_toSwiftConventions(name)),
```

Comment:
Hm, I actually *don't* like this change. It seems like it's not InputFile's job to know that the conventions are different.